### PR TITLE
Full feedbooks import

### PR DESCRIPTION
**Pull request by leonardr**
This branch adds specialization code to handle an OPDS import from Feedbooks's idiosyncratic OPDS feed. I have not done a full test of the Feedbooks monitor but I can correctly import records for individual books, whether they are redistributable in the US (in which case we mirror them) or not (in which case we don't mirror them).

There are two main problems that I solve:

* Feedbooks has open-access links with rel="http://opds-spec.org/acquisition". Vanilla OPDS importer ignores these links, so we modify the LinkData objects in place to have the correct link relation.

* It's impossible to determine the rights status of a book until we're deep in the Elementtree code, but the default rights status for the entry is fetched earlier on, in the Feedparser code. I have the feedparser code decline to provide a value, so that the elementtree code can come in later and fill in the value.

I also changed `can_rehost_us` slightly to return None in cases where we just don't know (without manual effort) whether we can rehost a book.

### CODE REVIEW

File: feedbooks.py [hunk 1]
```diff
@@ -29,23 +32,71 @@ def extract_feed_data(self, feed, feed_url=None):
...
+    def rights_uri_from_entry_tag(cls, entry):
+        rights = OPDSXMLParser._xpath1(entry, 'atom:rights')
+        if rights is not None:
+            rights = rights.text
+        source = OPDSXMLParser._xpath1(entry, 'dcterms:source')
+        if source is not None:
+            source = source.text
+        publication_year = OPDSXMLParser._xpath1(entry, 'dcterms:issued')
+        if publication_year is not None:
+            publication_year = publication_year.text
```
Comment (leonardr) [inline feedbooks.py]: This method is called by code in core OPDSImporter. We extract the information necessary to determine our rights to this book (as opposed to Feedbooks's rights), and run it through the preexisting `RehostingPolicy.rights_uri` method.

File: feedbooks.py [hunk 2]
```diff
@@ -29,23 +32,71 @@ def extract_feed_data(self, feed, feed_url=None):
             self.improve_description(id, m)
         return metadata, failures
 
-    def rights_for_entry(self, entry):
+    @classmethod
+    def rights_uri_from_feedparser_entry(cls, entry):
```
Comment (leonardr) [inline feedbooks.py]: This method is called by code in core OPDSImporter. As the comment says, it's not reliable, and the result needs to be fixed later, but we do the best we can.

File: feedbooks.py [hunk 3]
```diff
@@ -29,23 +32,71 @@ def extract_feed_data(self, feed, feed_url=None):
...
+
+        We have to correct the data after the fact instead of getting
+        it right within rights_uri_from_feedparser_entry, because
+        dcterms:issued (which we use to determine whether a work is
+        public domain in the United States) is not available through
+        Feedparser.
+        """
+        rights_uri = cls.rights_uri_from_entry_tag(entry_tag)
+        circulation = feedparser_detail.setdefault('circulation', {})
+        circulation['default_rights_uri'] = rights_uri
```
Comment (leonardr) [inline feedbooks.py]: Here's where we use Elementtree data to correct the Feedparser data.

File: tests/test_feedbooks.py [hunk 4]
```diff
@@ -43,12 +53,14 @@ def sample_file(self, filename):
...
+    def test_rights_uri_from_feedparser_entry(self):
         entry = dict(rights=LIFE_PLUS_70,
-                     source='gutenberg.net.au',
-                     publication_year="1922")
-        rights = self.importer.rights_for_entry(entry) 
-        eq_(RightsStatus.CC_BY_NC, rights)
+                     source='gutenberg.net.au')
+        expect = RehostingPolicy.rights_uri(
+            LIFE_PLUS_70, 'gutenberg.net.au', None
+        )
```
Comment (leonardr) [inline tests/test_feedbooks.py]: Had to add a test where publication_year is None because that's what happens when we try to determine the rights URI based on incomplete feedparser data.

**Comment by aslagle** [comment]
I made some comments on the core branch, but this all looks good.
